lib/OptimizationNLPModels/src/OptimizationNLPModels.jl

```diff
@@ -0,0 +1,38 @@
+module OptimizationNLPModels
+
+using Reexport
+@reexport using NLPModels, Optimization, ADTypes
+
+function SciMLBase.OptimizationFunction(
+        nlpmodel::AbstractNLPModel; kwargs...)
+    f(x, p) = NLPModels.obj(nlpmodel, x)
+    grad(G, u, p) = NLPModels.grad!(nlpmodel, u, G)
+    hess(u, p) = NLPModels.hess(nlpmodel, u)
```

Comment:
```suggestion
    hess(H, u, p) = NLPModels.hess!(nlpmodel, u, H)
```

Reply:
I think the in-place version of `hess` is not implemented (or at least not documented). I'm using the [API section of NLPModels](https://jso.dev/NLPModels.jl/latest/reference/) as a reference

Reply:
yeah can't say I understand it well either, we can just do this for such cases for now

```suggestion
    hess(H, u, p) = (H .= NLPModels.hess(nlpmodel, u))
```

Reply:
Why did you not make the change but resolve this comment? This will error since optimizers assume inplace functions for these